crates/core/tedge/src/cli/certificate/create.rs

```diff
@@ -55,20 +79,42 @@ impl CreateCertCmd {
         validate_parent_dir_exists(&self.cert_path).map_err(CertError::CertPathError)?;
         validate_parent_dir_exists(&self.key_path).map_err(CertError::KeyPathError)?;
 
-        let cert = KeyCertPair::new_selfsigned_certificate(config, &self.id, key_kind)?;
-
-        // TODO cope with broker user being tedge
-        // Creating files with permission 644 owned by the MQTT broker
-        let mut cert_file =
-            create_new_file(&self.cert_path, crate::BROKER_USER, crate::BROKER_GROUP)
-                .map_err(|err| err.cert_context(self.cert_path.clone()))?;
-
-        let cert_pem = cert.certificate_pem_string()?;
-        cert_file.write_all(cert_pem.as_bytes())?;
-        cert_file.sync_all()?;
-
-        // Prevent the certificate to be overwritten
-        set_permission(&cert_file, 0o444)?;
+        let cert = match &self.csr_path {
+            Some(csr_path) => {
+                validate_parent_dir_exists(csr_path).map_err(CertError::CsrPathError)?;
+
+                let cert = KeyCertPair::new_certificate_sign_request(config, &self.id, key_kind)?;
+                let cert_csr = cert.certificate_signing_request_string()?;
+
+                create_file_with_user_group(
+                    csr_path,
+                    crate::BROKER_USER,
+                    crate::BROKER_GROUP,
+                    0o444,
+                    Some(cert_csr.as_str()),
+                )?;
+
+                cert
+            }
+            None => {
+                let cert = KeyCertPair::new_selfsigned_certificate(config, &self.id, key_kind)?;
+
+                // TODO cope with broker user being tedge
+                // Creating files with permission 644 owned by the MQTT broker
+                let mut cert_file =
+                    create_new_file(&self.cert_path, crate::BROKER_USER, crate::BROKER_GROUP)
+                        .map_err(|err| err.cert_context(self.cert_path.clone()))?;
```

Comment:
This TODO is in the wrong place. It must be related to the private key, which owner matters.

Reply:
Yeah, I just resolved the conflict and placed this comment in the same spot as [here](https://github.com/thin-edge/thin-edge.io/blob/01f1cea1d7459e177c89f715c1a54b54d6c7034c/crates/core/tedge/src/cli/certificate/create.rs#L60)